### PR TITLE
Reorganize server files

### PR DESCRIPTION
**Pull request by mshima**
- reorganize by functionality.
- Move user to entity-files

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/app/__snapshots__/generator.spec.mts.snap [hunk 1]
```diff
@@ -793,6 +794,7 @@ exports[`generator - app with gateway should match snapshot 1`] = `
   "gatlingTests": false,
   "generateBuiltInAuthorityEntity": true,
   "generateBuiltInUserEntity": true,
+  "generateLoginManagement": true,
```
Comment (mraible) [inline generators/app/__snapshots__/generator.spec.mts.snap:797]: What is login management in this context?

Comment (mshima) [reply generators/app/__snapshots__/generator.spec.mts.snap:797]: Authentication api. Any suggestion?

Comment (mshima) [reply generators/app/__snapshots__/generator.spec.mts.snap:797]: `generateAuthenticationApi`?

Comment (mraible) [reply generators/app/__snapshots__/generator.spec.mts.snap:797]: Yes, I think that makes more sense.

File: generators/server/files.mjs [hunk 2]
```diff
@@ -17,9 +17,245 @@
...
+      templates: ['test/util/OAuth2TestUtil.java', 'web/rest/LogoutResourceIT.java'],
+    },
+    {
+      condition: generator => !generator.reactive && generator.generateLoginManagement,
+      path: `${SERVER_MAIN_SRC_DIR}package/`,
+      renameTo: moveToJavaPackageSrcDir,
+      templates: ['security/oauth2/CustomClaimConverter.java'],
+    },
+    {
+      condition: generator => !generator.reactive && generator.generateLoginManagement,
```
Comment (DanielFran) [inline generators/server/files.mjs:178]:
```suggestion
      condition: generator => !generator.reactive && generator.generateAuthenticationApi,
```

File: generators/server/files.mjs [hunk 3]
```diff
@@ -40,35 +276,27 @@ const jwtFiles = {
...
+      condition: data => !data.generateLoginManagement,
       path: `${SERVER_TEST_SRC_DIR}package/`,
       renameTo: moveToJavaPackageTestDir,
       templates: ['security/jwt/TestAuthenticationResource.java'],
     },
-  ],
-  entrypointFiles: [
     {
-      condition: generator => !generator.applicationTypeMicroservice,
+      condition: generator => generator.generateLoginManagement,
```
Comment (DanielFran) [inline generators/server/files.mjs:288]:
```suggestion
      condition: generator => generator.generateAuthenticationApi,
```

File: generators/server/files.mjs [hunk 4]
```diff
@@ -40,35 +276,27 @@ const jwtFiles = {
...
       path: `${SERVER_MAIN_SRC_DIR}package/`,
       renameTo: moveToJavaPackageSrcDir,
       templates: ['web/rest/vm/LoginVM.java', 'web/rest/AuthenticateController.java'],
     },
     {
-      condition: generator => !generator.applicationTypeMicroservice,
+      condition: generator => generator.generateLoginManagement,
```
Comment (DanielFran) [inline generators/server/files.mjs:294]:
```suggestion
      condition: generator => generator.generateAuthenticationApi,
```

File: generators/server/files.mjs [hunk 5]
```diff
@@ -702,55 +629,33 @@ export const baseServerFiles = {
...
+  ...commonUserFiles,
 };
 
 export const serverFiles = mergeSections(
   baseServerFiles,
   addSectionsCondition(jwtFiles, context => context.authenticationTypeJwt),
-  addSectionsCondition(gatewayFiles, context => context.applicationTypeGateway)
+  addSectionsCondition(oauth2Files, context => context.authenticationTypeOauth2),
+  addSectionsCondition(gatewayFiles, context => context.applicationTypeGateway),
+  addSectionsCondition(accountFiles, context => context.generateLoginManagement),
```
Comment (DanielFran) [inline generators/server/files.mjs:646]:
```suggestion
  addSectionsCondition(accountFiles, context => context.generateAuthenticationApi),
```

File: generators/server/generator.mjs [hunk 6]
```diff
@@ -370,6 +370,7 @@ export default class JHipsterServerGenerator extends BaseApplicationGenerator {
         application.frontendAppName = this.getFrontendAppName(application.baseName);
         application.mainClass = this.getMainClassName(application.baseName);
         application.authenticationUsesCsrf = [OAUTH2, SESSION].includes(application.authenticationType);
+        application.generateLoginManagement = application.applicationTypeMonolith || application.applicationTypeGateway;
```
Comment (DanielFran) [inline generators/server/generator.mjs:373]:
```suggestion
        application.generateAuthenticationApi = application.applicationTypeMonolith || application.applicationTypeGateway;
```

**Review by DanielFran — APPROVED** [review]

**Pull request merged**